### ISSUE TITLE
website application error

### ISSUE THREAD

**Issue by juangomes376** [matteo-convertino/otpmanager-nextcloud#11]
First of all, congratulations on the initiative and the phone app is already working well, but I don't know if it's me that's not doing it right, but my phone app can't communicate with the website, I tried to install it again from the store and it's blocked on the loading, I tried to install it manually and it doesn't work either, and one thing that catches my attention is the number of files is not the same for both, if you need help I'm a developer too and I can try to help.
![Capture d’écran 2023-10-22 010017](https://github.com/matteo-convertino/otpmanager-nextcloud/assets/62665231/a08868cf-784e-4272-b40a-104b65c5932b)

**Comment by juangomes376** [comment]
observation, I don't even know how to try to solve it because it doesn't give any error, I looked in the server log and in the nextcloud log itself

**Comment by matteo-convertino** [comment]
Hi Juan,
It seems like this is a problem other users are experiencing as well (#10).
The fact that there aren't even any log messages complicates the issue because I can't reproduce the problem.

When you see this infinite loading, don't any errors appear in the console? If you go to the "network" section, do you see any requests that have a status code of 500 or similar?

**Comment by felsss** [comment]
Hello,

Resuming from my last message in #10 
I have 2 404 errors in the web console, see screenshot below, and full HAR can be provided if needed.

![image](https://github.com/matteo-convertino/otpmanager-nextcloud/assets/46402789/80c78fb4-ac00-42af-8d8b-62ac3b2a9639)

**Comment by matteo-convertino** [comment]
The fact that they return 404 is very strange.

At this point I have a doubt: which URL were the two requests "password" and "settings" made to?
For example it was `https://flxrch.eu/owncloud/apps/otpmanager/settings` or it is different (e.g. ignore the `owncloud` "subdirectory").

You could also try to manually go to `https://flxrch.eu/owncloud/apps/otpmanager/settings` and see the result.

**Comment by juangomes376** [comment]
in mine the network error is in the two referenced links, but in the console there are many jquery errors, later I can try to see if I can correct some of these errors.
![Capture d’écran du 2023-10-23 09-09-04](https://github.com/matteo-convertino/otpmanager-nextcloud/assets/62665231/85740e2c-483a-4e0c-a3d3-3b4f38ae0925)

**Comment by felsss** [comment]
Hello,
On my side, both 404 were requests that ignore the subdirectory : 
![image](https://github.com/matteo-convertino/otpmanager-nextcloud/assets/46402789/7d77489c-31b9-4767-ba10-30da726a8589)
Manuel request to the same URL with subdirectory included does not send a 404.

**Comment by matteo-convertino** [comment]
> Hello, On my side, both 404 were requests that ignore the subdirectory : ![image](https://user-images.githubusercontent.com/46402789/277263677-7d77489c-31b9-4767-ba10-30da726a8589.png) Manuel request to the same URL with subdirectory included does not send a 404.

Perfect, then in your case the problem is the generation of the URL. I try to fix it as soon as possible.

@juangomes376 Yours seems like a different problem though. You said earlier that you had also tried installing the extension manually. It is still so? Because obviously I recommend you install it directly from the store.

**Comment by juangomes376** [comment]
Now you have the application installed from the store, but before I tried to install it manually.
Then it didn't work, I deleted everything and installed it from the store.
But I'm having this problem with the links too.
![Capture d’écran du 2023-10-23 13-17-51](https://github.com/matteo-convertino/otpmanager-nextcloud/assets/62665231/30916903-6b0b-4ce1-bf73-a161c81675ae)

I thought the problem with these links could be due to jquery errors

**Comment by matteo-convertino** [comment]
Ok so I assume that you also have nextcloud with a subfolder `www.example.com/subfolder` but the request is made on `www.example.com` instead, so it returns 404. Is that right?

**Comment by juangomes376** [comment]
No, my server points directly to the nextcloud folder

**Comment by matteo-convertino** [comment]
I just published version [0.3.1](https://github.com/matteo-convertino/otpmanager-nextcloud/releases/tag/v0.3.1) where this error should be fixed. However, I don't know how long after it will appear on the nextcloud store (due to cache).
Let me know if the problem is resolved for both of you.

**Comment by felsss** [comment]
Hello,
Just updated web to 0.3.1 and android app to 1.3.1.15.
Web app correctly prompted for password creation.
I added a new totp account in web app, configured the password in android app, and sync went OK.
I'll go on with more testing, but everything seems to work fine for my setup now.
Thanks for the quick handling of the issue !

**Comment by juangomes376** [comment]
For me, the website is working but my app can't connect, but this part could be due to my old phone.
it is blocked on the password entry page.

**Comment by matteo-convertino** [comment]
Do you get any error messages when you try to enter the password (just for clarity, the password would be the one you create the first time you log in from the web app)?

**Comment by juangomes376** [comment]
I looked at the server logs and didn't see anything wrong. but I can see that the mobile app checks the server. But that's what it looks like for me.
so I assume it's my phone.

error.log null

nextcloud.log null

accesses.log [24/Oct/2023:19:43:54 +0000] "POST /index.php/apps/otpmanager/password/check HTTP/1.1" 401 6357 "-" "Dart/3.0 (dart:io)"

**Comment by juangomes376** [comment]
I would like to thank everyone I just found out who my phone is.
If I look at the password check directly on the website it has 3 different urls.
1 .../check
2 ..../accounts
3 ...../settings
and when I look at the traffic on the server when I send the password over the phone I only have .../check and there are no other 2. I have to find a way to see what my phone is sending to analyze it better. But if you want to close the topic, I'd appreciate your help and I hope I helped you resolve this part of the system. and in the future, if you need people to test things, you can call me.

**Comment by matteo-convertino** [comment]
Thank you and I am glad you like the project (testers/feedback are always appreciated :smile:).

> accesses.log [24/Oct/2023:19:43:54 +0000] "POST /index.php/apps/otpmanager/password/check HTTP/1.1" 401 6357 "-" "Dart/3.0 (dart:io)"

However, I would have doubts about the 401 status code that returns for that request in the accesses.log. Most likely you don't see /accounts because /checks returns a status code other than 200.
Pretty strange though. Have you tried logging out and then logging back in to the app?

Since the reason for opening this issue is resolved, I'll close it. If you need a hand, please open another issue for this other problem so I can help you.
